### PR TITLE
bump PMDK to v 1.7

### PR DESCRIPTION
**Pull request by carns**
Update builtin repo with latest version of PMDK.

### CODE REVIEW

File: var/spack/repos/builtin/packages/pmdk/package.py [hunk 1]
```diff
@@ -18,6 +18,7 @@ class Pmdk(Package):
     git      = "https://github.com/pmem/pmdk.git"
 
     version('develop', branch='master')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/pmdk/package.py]: If you want you can actually rename this version to `master` now that #1983 has been merged.

Comment (carns) [reply var/spack/repos/builtin/packages/pmdk/package.py]: @adamjstewart what is the recommended convention now?  I don't have a preference on the name myself, other than it is inconvenient to remember which packages are using `master` and which are using `develop` for this sort of thing.  If the consensus is to use `master` for the latest master branch then I'll push an update to this PR.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/pmdk/package.py]: There isn't really a consensus. My personal opinion is that we shouldn't be renaming branches that the developers themselves named. But I also never install anything from master/develop so I've never had to remember which package uses which name.

**Comment by alalazo** [comment]
@carns Feel free to submit the modification suggested above in another PR if it makes sense to you.

**Comment by carns** [comment]
Thanks all.  I'll come back with a PR for the version naming, we need to think about that in multiple packages probably.

**Pull request merged**